pkg/aspect/types.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aspect
+
+import (
+	"io"
+
+	"github.com/golang/protobuf/proto"
+	"google.golang.org/genproto/googleapis/rpc/code"
+
+	"istio.io/mixer/pkg/attribute"
+	"istio.io/mixer/pkg/expr"
+
+	istioconfig "istio.io/api/istio/config/v1"
+)
+
+type (
+	// Adapter represents a factory to create an adapterImpl that provides a specific aspect
```

Comment:
I don't think this sentence is quite right: The developer creates an adapter implementation, not a factory method. Perhaps:

Adapter represents the factory responsible for returning instantiations of particular aspects.